src/main/java/io/lettuce/core/ClientOptions.java

```diff
@@ -354,6 +373,18 @@ public TimeoutOptions getTimeoutOptions() {
         return timeoutOptions;
     }
 
+    /**
+     * Buffer usage ratio for {@link io.lettuce.core.protocol.CommandHandler}. Discard operation for read bytes occur
+     * When buffer usage reach {@code bufferUsageRatio / bufferUsageRatio + 1}. For example, sets 3 to bufferUsageRatio,
+     * means buffer usage reach 75 percentage, discard operation occur.
+     *
+     * @return the buffer usage ratio.
+     * @since 5.2
+     */
+    public int getBufferUsageRatio() {
```

Comment:
I'd propose to make cleaning more customizable. This method could return a discard consumer (`BiConsumer<ByteBuf, DecodeProgress>`) instead of exposing details how discard of buffer bytes is configured.

See below for the full explanation.